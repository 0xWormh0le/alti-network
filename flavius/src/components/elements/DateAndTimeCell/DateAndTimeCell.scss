@import 'styles/_shared.scss';

.DateAndTimeCell {
  min-width: 83px;
  max-width: 300px;

  &__date,
  &__time {
    display: block;
  }

  &__time {
    @include set-typography('BODY_SMALL', 'normal');
    color: $TEXT_COLOR_SECONDARY;
  }

  &__time-container {
    display: flex;
    gap: $BASE_SPACING;
  }

  &__date--unknown {
    font-style: oblique;
    color: $TEXT_COLOR_SECONDARY;
  }
}
