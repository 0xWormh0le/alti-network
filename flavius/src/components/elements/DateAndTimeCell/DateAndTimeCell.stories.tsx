import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DateAndTimeCell, { DateAndTimeCellProps } from './DateAndTimeCell'

export default {
  title: 'Elements/DateAndTimeCell',
  component: DateAndTimeCell
} as Meta

const Template: Story<DateAndTimeCellProps> = (args) => <DateAndTimeCell {...args} />

export const Default = Template.bind({})
Default.args = {
  value: 1
}
