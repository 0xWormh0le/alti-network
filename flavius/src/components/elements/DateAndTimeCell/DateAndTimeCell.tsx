import React from 'react'
import Moment from 'react-moment'
import CONSTANTS from 'util/constants'
import './DateAndTimeCell.scss'

const { TIME_DISPLAY_FORMAT } = CONSTANTS

export interface DateAndTimeCellProps {
  value: number
}

export const DateAndTimeCell: React.FC<DateAndTimeCellProps> = ({ value, children }) => (
  <div className='DateAndTimeCell'>
    {!value ? (
      <span className='DateAndTimeCell__date--unknown'>unknown</span>
    ) : (
      <React.Fragment>
        <Moment unix={true} format={TIME_DISPLAY_FORMAT.DATE_FORMAT} className='DateAndTimeCell__date'>
          {value}
        </Moment>
        <div className='DateAndTimeCell__time-container'>
          <Moment className='DateAndTimeCell__time' unix={true} format={TIME_DISPLAY_FORMAT.TIME_FORMAT}>
            {value}
          </Moment>
          {children}
        </div>
      </React.Fragment>
    )}
  </div>
)

export default DateAndTimeCell
