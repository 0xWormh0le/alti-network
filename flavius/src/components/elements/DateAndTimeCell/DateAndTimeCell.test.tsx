import React from 'react'
import { render } from '@testing-library/react'
import { DateAndTimeCell, DateAndTimeCellProps } from './DateAndTimeCell'

describe('DateAndTimeCell', () => {
  it('renders correctly', () => {
    const props: DateAndTimeCellProps = {
      value: 1566492823
    }
    const { container } = render(<DateAndTimeCell {...props} />)
    expect(container).toMatchSnapshot()
  })
})
