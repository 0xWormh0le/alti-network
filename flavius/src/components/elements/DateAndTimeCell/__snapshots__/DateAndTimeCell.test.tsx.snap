// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateAndTimeCell renders correctly 1`] = `
<div>
  <div
    class="DateAndTimeCell"
  >
    <time
      class="DateAndTimeCell__date"
      datetime="1566492823000"
    >
      Aug 22, 2019
    </time>
    <div
      class="DateAndTimeCell__time-container"
    >
      <time
        class="DateAndTimeCell__time"
        datetime="1566492823000"
      >
        09:53am
      </time>
    </div>
  </div>
</div>
`;
